crypto/batchverifier.go

```diff
@@ -111,6 +112,18 @@ func (b *BatchVerifier) getNumberOfEnqueuedSignatures() int {
 // Verify verifies that all the signatures are valid. in that case nil is returned
 // if the batch is zero an appropriate error is return.
 func (b *BatchVerifier) Verify() error {
+	return b.VerifyWithFeedback(nil)
+}
+
+// VerifyWithFeedback verifies that all the signatures are valid.
+// failed slice should have len equal to getNumberOfEnqueuedSignatures
+// if all sigs are valid, nil will be returned
+// if some txns are invalid, true will be set at the appropriate index in failed
+func (b *BatchVerifier) VerifyWithFeedback(failed []bool) error {
+	if failed != nil && len(failed) != b.getNumberOfEnqueuedSignatures() {
```

Comment:
`failed != nil` is not needed. `len(failed)` will return 0 in that case.

Reply:
I think this is confusing. I need to refactor this. thanks for the comment.